Sources/Algorithms/Combinations.swift

```diff
@@ -14,17 +14,54 @@ public struct Combinations<Base: Collection> {
   /// The collection to iterate over for combinations.
   public let base: Base
   
+  /// The range of accepted sizes of combinations.
+  /// - Note: This may be `nil` if the attempted range entirely exceeds the
+  /// upper bounds of the size of the `base` collection.
   @usableFromInline
-  internal var k: Int
+  internal let kRange: Range<Int>?
   
+  /// Initializes a `Combinations` for all combinations of `base` of all sizes.
+  /// - Parameter base: The collection to iterate over for combinations.
+  @usableFromInline
+  internal init(_ base: Base) {
+    self.init(base, kRange: 0...)
+  }
+  
```

Comment:
```suggestion  
```

I don't think this is being used anymore

Reply:
Thanks! I’ve now removed it from the original commit making the change to remove `combinations()`.